libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/preferences/PreferenceSwitch.kt

```diff
@@ -53,62 +42,57 @@ fun PreferenceSwitch(
     icon: ImageVector? = null,
     @DrawableRes iconResourceId: Int? = null,
     showIconAreaIfNoIcon: Boolean = false,
-    switchAlignment: Alignment.Vertical = Alignment.CenterVertically
 ) {
-    Row(
-        modifier = modifier
-            .fillMaxWidth()
-            .defaultMinSize(minHeight = preferenceMinHeight)
-            .clickable { onCheckedChange(!isChecked) }
-            .padding(vertical = 4.dp, horizontal = preferencePaddingHorizontal),
-        verticalAlignment = Alignment.CenterVertically
-    ) {
-        PreferenceIcon(
+    ListItem(
+        modifier = modifier,
+        enabled = enabled,
+        onClick = onCheckedChange.takeIf { enabled }?.let { { onCheckedChange(!isChecked) } },
+        leadingContent = preferenceIcon(
             icon = icon,
             iconResourceId = iconResourceId,
             enabled = enabled,
-            isVisible = showIconAreaIfNoIcon
-        )
-        Column(
-            modifier = Modifier
-                .weight(1f)
-                .align(Alignment.CenterVertically)
-        ) {
+            showIconAreaIfNoIcon = showIconAreaIfNoIcon,
+        ),
+        headlineContent = {
             Text(
                 style = ElementTheme.typography.fontBodyLgRegular,
                 text = title,
-                color = enabled.toEnabledColor(),
             )
+        },
+        supportingContent = {
```

Comment:
This should probably be `supportingContent = subtitle?.let { ... }`, I think otherwise the preference will add some spacing that's not really needed.

Reply:
Yes, having an empty Composable does not seems to have an impact, but let's do what you're suggesting to be sure.